src/Stack.cpp

```diff
@@ -0,0 +1,153 @@
+#pragma once
+#include "Stack.h"
+
+
+Stack::Stack(const size_t& size, const int& value)
+	:Vector::Vector(size, value)
+{
+}
+
+// Copy constructor
+Stack::Stack(const Stack& other) {
+	*this = other;//use Vector Copy Assiment
+}
+
+// Move Constructor
+Stack::Stack(Stack&& other) noexcept
+{
+	Vector::operator=(other);
+}
+
+Stack::~Stack()
+{
+}
+
+/* Assimengt */
+Stack& Stack::operator=(const Stack& other)
+{
+	Vector::operator=(other);
+	return *this;
+}
+
+Stack& Stack::operator=(Stack&& other) noexcept
+{
+	Vector::operator=(other);
+	return *this;
+}
+
+int Stack::top()
+{
+	size_t index = size() - 1;
+	if (index < 0 || index >= size()) {
+		throw std::out_of_range("Index out of bounds");
+	}
+	return (*this)[index];
```

Comment:
Why use the `operator[]` which again does the same `if` you just did two lines above?